content/configuring/anomaly_scoring.md

```diff
@@ -0,0 +1,151 @@
+
+## How Anomaly Scoring Mode Works
+
+Anomaly scoring mode combines the concepts of *collaborative detection* and *delayed blocking*. The key idea to understand is that **the inspection/detection rule logic is decoupled from the blocking functionality**.
+
+Individual rules designed to detect specific types of attacks and malicious behaviour are executed. If a rule matches, no immediate disruptive action is taken (e.g. the transaction is not blocked). Instead, the matched rule contributes to a transactional *anomaly score*, which acts as a running total. The rules just handle detection, adding to the anomaly score if they match. In addition, an individual matched rule will typically log a record of the match for later reference, including the ID of the matched rule, the data that caused the match, and the URI that was being requested.
+
+Once all of the rules that inspect *request* data have been executed, *blocking evaluation* takes place. If the anomaly score is greater than or equal to the inbound anomaly score threshold then the transaction is *denied*. Transactions that are not denied continue on their journey.
+
+![Diagram showing an example where the inbound anomaly score threshold is set to 5. A first example request accumulates an anomaly score of 7 and is denied at the blocking evaluation step. A second example request accumulates no anomaly score and is allowed to pass at the blocking evaluation step.](as_inbound_no_fonts.svg?height=36em)
+
+Continuing on, once all of the rules that inspect *response* data have been executed, a second round of blocking evaluation takes place. If the *outbound* anomaly score is greater than or equal to the outbound anomaly score threshold then the transaction is *denied*.
+
+{{% notice info %}}
+Having separate inbound and outbound anomaly scores and thresholds allows for request data and response data to be inspected and scored independently.
+{{% /notice %}}
+
+### Summary of Anomaly Scoring Mode
+
+To summarise, anomaly scoring mode in the CRS works like so:
+
+1. Execute all *request* rules
+1. Make a blocking decision using the *inbound* anomaly score threshold
+1. Execute all *response* rules
+1. Make a blocking decision using the *outbound* anomaly score threshold
+
+### The Anomaly Scoring Mechanism In Action
+
+As described, individual rules are only responsible for detection and inspection: they do not block or deny transactions. If a rule matches then it increments the anomaly score. This is done using ModSecurity's `setvar` action.
+
+Below is an example of a detection rule which matches when a request contains an empty `User-Agent` header. Notice the final line of the rule: it makes use of the `setvar` action, which will increment the anomaly score if the rule matches:
+
+```apache
+SecRule REQUEST_HEADERS:User-Agent "@rx ^$" \
```

Comment:
This is unfortunately a suboptimal example, since it's a notice-rule that carries "pass" as the blocking action. It would be better to have a critical rule that comes with "block".

Reply:
Good point. I've changed this to a `CRITICAL` severity rule featuring the `block` action.